### ISSUE TITLE
bricknames and targets

### ISSUE THREAD

**Issue by ngbusca** [desihub/redrock#29]
l. 187 in external/desi.py asserts len(brickname)==len(targets)

My run on plate 4199 of boss2desi is crashing at this line since I have one target rejected due to

```
ERROR: Target 102696313 on brick-b-4199.fits has no good spectra
```

and the assertion fails. I suggest moving l.177 of desi.py inside the if statement in l. 180 to avoid the crash.

**Comment by ngbusca** [comment]
With this change, the code runs smoothly (although I haven't checked whether this change breaks the bookkeeping somewhere else).

**Comment by sbailey** [comment]
fixed by PR #30
